### DISCUSSION TITLE
Adding a custom model - day and night license plate recognition

### DISCUSSION THREAD

**Discussion by inspired27** [blakeblackshear/frigate#6822] [Ask A Question]
I am using docker version of Frigate and want to incorporate this excellent day and night license plate model by Mike Lud into Frigate can anyone help how to include to accompany the inbuilt models. For example which directory etc.

https://github.com/MikeLud/CodeProject.AI-Custom-IPcam-Models/tree/main/PT%20Models

**Comment by NickM-27** [comment]
Currently only one model is supported, you can't run multiple at the same time

**Reply by inspired27** [reply]
Okay, is it possible to merge in this license plate model into the frigate model if so how?

**Reply by NickM-27** [reply]
I don't believe so, but of course the current Frigate model is just a Google demo model so you can download it and try to merge yourself. 

The Frigate+ model already includes license plate as a label (though it's not out yet)

**Reply by inspired27** [reply]
When do you anticipate it being released? I would be happy to donate to accelerate the license plate recognition. Mike Lud dataset might help. Crime is happening regularly in our area with car theft, never used to be but is now and I need license plate recognition.

**Reply by NickM-27** [reply]
To be clear the model does license plate detection, ALPR is separate. I don't work on Frigate+ but I believe it was said on the HA discord taht the models will be generally available alongside 0.13

**Reply by inspired27** [reply]
Oh good that it at least identifies plates that's what I want Day and Night plate object detection then I will get home assistant to process the image thereafter. Is it available in the 0.13 dev version you put out for Cody's issue if so I will use the dev version and pay for Frigate+

**Reply by NickM-27** [reply]
No, it is separate.

**Comment by inspired27** [comment]
Thanks Nic, with Frigate+ when signed up is it all stored local in the docker container or does Frigate call out to the Frigate+ repository everytime to recognise objects?

If local I am super keen to try it out are there any instructions to get up and running as license plate recognition more so at night is huge part of my use case.

________________________________

**Reply by NickM-27** [reply]
It runs locally just like any other model

**Comment by inspired27** [comment]
Okay so I finally figured it out, it is possible to add custom model via the deepstack detector plugin in. I am running dev 0.13.0-0996883 with deepstack plugin pointing to codeproject.ai server which is running okay. However the license-plate model in the codeproject.ai server results in return of **Dayplate** or **Nightplate** with capitalise first letter and is not being passed back to Frigate. Please can someone confirm what I need to add to the Frigate config file (and if capitalised or no caps or both), and do I need to update labelmap.txt if so with what?

```
detectors:
  deepstack:
    api_url: http://192.168.1.xx:32168/v1/vision/custom/license-plate
    type: deepstack
    api_timeout: 1 # seconds
```

**Comment by kirsch33** [comment]
> Okay so I finally figured it out, it is possible to add custom model via the deepstack detector plugin in. I am running dev 0.13.0-0996883 with deepstack plugin pointing to codeproject.ai server which is running okay. However the license-plate model in the codeproject.ai server results in return of **Dayplate** or **Nightplate** with capitalise first letter and is not being passed back to Frigate. Please can someone confirm what I need to add to the Frigate config file (and if capitalised or no caps or both), and do I need to update labelmap.txt if so with what?
> 
> ```
> detectors:
>   deepstack:
>     api_url: http://192.168.1.xx:32168/v1/vision/custom/license-plate
>     type: deepstack
>     api_timeout: 1 # seconds
> ```
see this comment:

https://github.com/blakeblackshear/frigate/pull/6143#issuecomment-1518474242
